app.py

```diff
@@ -31,20 +31,20 @@ def post(self):
         input_lang = args['in_lang']
         output_lang = args['out_lang']
 
-        output_code = main(input_code, input_lang, output_lang)
-        response_input_lang = input_lang
-
-        # Gets non-beta (fully supported) languages for automatic detection
-        fully_supported_lang_codes = ConverterRegistry.get_fully_supported_language_codes(
-        )
-
-        # automatic language detection (only fully supported languages) TODO: fall back on Beta if all else fails
-        i = 0
-        while (output_code == "Error: did not compile") and (
-                i < len(fully_supported_lang_codes)):
-            response_input_lang = fully_supported_lang_codes[i]
-            output_code = main(input_code, response_input_lang, output_lang)
-            i += 1
+        if input_lang == "auto":
+            # Gets non-beta (fully supported) languages for automatic detection
+            fully_supported_lang_codes = ConverterRegistry.get_fully_supported_language_codes(
+            )
+            # automatic language detection (only fully supported languages) TODO: fall back on Beta if all else fails
+            for lang in fully_supported_lang_codes:
+                response_input_lang = lang
+                output_code = main(input_code, response_input_lang, output_lang)
+
+                if "Error" not in output_code:
```

Comment:
will it ever be the case no languages compile and is there an error message for that?

Reply:
Yes, once the last language doesn't compile, it will just return "Error: could not compile"